runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -211,6 +211,355 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, labelStringNotFound);
+   generateRIInstruction(cg, TR::InstOpCode::LHI, node, resultReg, -1);
+
+   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, labelResultDone);
+
+   // START result debug counters
+   if (enableIndexOfDebugCounter)
+      {
+      TR::LabelSymbol* labelResultGT10      = generateLabelSymbol(cg);
+      TR::LabelSymbol* labelResultGT30      = generateLabelSymbol(cg);
+      TR::LabelSymbol* labelResultGT60      = generateLabelSymbol(cg);
+      TR::LabelSymbol* labelResultGT100     = generateLabelSymbol(cg);
+      TR::LabelSymbol* labelResultCheckDone = generateLabelSymbol(cg);
+
+      uint8_t boundary10Char  = 10;
+      uint8_t boundary30Char  = 30;
+      uint8_t boundary60Char  = 60;
+      uint8_t boundary100Char = 100;
+
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpImmBranchRelOpCode(), node, s2LenReg, boundary10Char, labelResultGT10, TR::InstOpCode::COND_BH);
+      cg->generateDebugCounter("indexOfString/result/below-10", 1, TR::DebugCounter::Cheap);
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, labelResultCheckDone);
+
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node,  labelResultGT10);
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpImmBranchRelOpCode(), node, s2LenReg, boundary30Char, labelResultGT30, TR::InstOpCode::COND_BH);
+      cg->generateDebugCounter("indexOfString/result/10-30", 1, TR::DebugCounter::Cheap);
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, labelResultCheckDone);
+
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node,  labelResultGT30);
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpImmBranchRelOpCode(), node, s2LenReg, boundary60Char, labelResultGT60, TR::InstOpCode::COND_BH);
+      cg->generateDebugCounter("indexOfString/result/30-60", 1, TR::DebugCounter::Cheap);
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, labelResultCheckDone);
+
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node,   labelResultGT60);
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpImmBranchRelOpCode(), node, s2LenReg, boundary100Char, labelResultGT100, TR::InstOpCode::COND_BH);
+      cg->generateDebugCounter("indexOfString/result/60-100", 1, TR::DebugCounter::Cheap);
+      generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, labelResultCheckDone);
+
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, labelResultGT100);
+      cg->generateDebugCounter("indexOfString/result/above-100", 1, TR::DebugCounter::Cheap);
+
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, labelResultCheckDone);
+      }
+   // END debug counters
+
+   /************************************** Done ******************************************/
```

Comment:
I don't think these comments are particularly useful given we have scoped blocks encapsulating the debug counter code and we have a done label right below.